x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/snapshotlifecycle/history/SnapshotCreationHistoryItem.java

```diff
@@ -0,0 +1,139 @@
+
+    private final Map<String, Object> snapshotConfiguration;
+    @Nullable
+    private final String errorDetails;
+
+    static final ParseField SNAPSHOT_CONFIG = new ParseField("configuration");
+    static final ParseField ERROR_DETAILS = new ParseField("error_details");
+
+    @SuppressWarnings("unchecked")
+    private static final ConstructingObjectParser<SnapshotCreationHistoryItem, String> PARSER =
+        new ConstructingObjectParser<>("snapshot_lifecycle_history_item", true,
+            (a, id) -> {
+                final long timestamp = (long) a[0];
+                final String policyId = (String) a[1];
+                final String repository = (String) a[2];
+                final String snapshotName = (String) a[3];
+                final String operation = (String) a[4];
+                final boolean success = (boolean) a[5];
+                final Map<String, Object> snapshotConfiguration = (Map<String, Object>) a[6];
+                final String errorDetails = (String) a[7];
+                return new SnapshotCreationHistoryItem(timestamp, policyId, repository, snapshotName, operation, success,
+                    snapshotConfiguration, errorDetails);
+            });
+
+    static {
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), TIMESTAMP);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), POLICY_ID);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), REPOSITORY);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), SNAPSHOT_NAME);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), OPERATION);
+        PARSER.declareBoolean(ConstructingObjectParser.constructorArg(), SUCCESS);
+        PARSER.declareObject(ConstructingObjectParser.constructorArg(), (p, c) -> p.map(), SNAPSHOT_CONFIG);
+        PARSER.declareStringOrNull(ConstructingObjectParser.constructorArg(), ERROR_DETAILS);
+    }
+
+    SnapshotCreationHistoryItem(long timestamp, String policyId, String repository, String snapshotName, String operation,
+                                       boolean success, Map<String, Object> snapshotConfiguration, String errorDetails) {
+        super(timestamp, policyId, repository, snapshotName, operation, success);
+        this.snapshotConfiguration = Objects.requireNonNull(snapshotConfiguration);
+        this.errorDetails = errorDetails;
+    }
+
+    public static SnapshotCreationHistoryItem successRecord(long timestamp, String policyId, CreateSnapshotRequest request,
+                                                            Map<String, Object> snapshotConfiguration) {
+        return new SnapshotCreationHistoryItem(timestamp, policyId, request.repository(), request.snapshot(), CREATE_OPERATION, true,
+            snapshotConfiguration, null);
+    }
+
+    public static SnapshotCreationHistoryItem failureRecord(long timeStamp, String policyId, CreateSnapshotRequest request,
```

Comment:
Maybe since we have the policy in `SnapshotLifecycleTask`, we should just pass in the time, `policy`, and attempted snapshot name here and derive the other parts from it?

Reply:
Good idea! I've switched these two methods to use the policy.